core/actions/index.ts

```diff
@@ -119,4 +157,56 @@ export class LegacyConfigConverter {
     }
     return legacyConfig;
   }
+
+  public static insertLegacyBigQueryOptionsToConfigProto<
+    T extends ILegacyTableConfig | ILegacyIncrementalTableConfig
+  >(legacyConfig: T): T {
+    if (legacyConfig?.bigquery) {
```

Comment:
nit: you can consider early exit to avoid nested ifs
`if (!legacyConfig?.bigquery) {
        return legacyConfig;
    }`